plugin-test/src/test/groovy/org/gradle/github/dependency/extractor/SingleProjectDependencyExtractorTest.groovy

```diff
@@ -149,6 +149,75 @@ class SingleProjectDependencyExtractorTest extends BaseExtractorTest {
         }
     }
 
+    def "build with transitive dependency updated by constraint"() {
+        given:
+        def bar10 = mavenRepo.module("org.test", "bar", "1.0").publish()
+        def bar11 = mavenRepo.module("org.test", "bar", "1.1").publish()
+        def foo = mavenRepo.module("org.test", "foo", "1.0").dependsOn(bar10).publish()
+        singleProjectBuildWithDependencies """
+        dependencies {
+            implementation "org.test:foo:1.0"
+            
+            constraints {
+                implementation "org.test:bar:1.1"
+            }
+        }
+        """
+        when:
+        succeeds("dependencies", "--configuration", "runtimeClasspath")
+
+        then:
+        def manifest = jsonManifest("project :")
+        (manifest.file as Map).source_location == "build.gradle"
+        def resolved = manifest.resolved as Map
+        resolved.keySet() == ["org.test:foo:1.0", "org.test:bar:1.1"] as Set
+        verifyAll(resolved["org.test:foo:1.0"] as Map) {
+            package_url == purlFor(foo)
+            relationship == "direct"
+            dependencies == ["org.test:bar:1.1"]
+        }
+        verifyAll(resolved["org.test:bar:1.1"] as Map) {
+            package_url == purlFor(bar11)
+            relationship == "direct" // Constraint is a type of direct dependency
+            dependencies == []
+        }
```

Comment:
There is a lot going on inside of these `then` blocks and I'm getting a little lost. It's not clear to me what they are verifying.

Reply:
At a high level, they are verifying the output JSON that is being produced and dumped out has the correct structure.

The schema for the JSON can be found here: https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28

Reply:
I'll cleanup the tests (to make them more readable) in a separate PR